libsolutil/CommonData.h

```diff
@@ -38,39 +38,45 @@
 /// Operators need to stay in the global namespace.
 
 /// Concatenate the contents of a container onto a vector
-template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U const& _b)
+template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U& _b)
 {
 	for (auto const& i: _b)
 		_a.push_back(i);
 	return _a;
 }
 /// Concatenate the contents of a container onto a vector, move variant.
-template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U&& _b)
+template <class T, class U,
+	class = std::enable_if_t<!std::is_lvalue_reference_v<U>>>
+std::vector<T>& operator+=(std::vector<T>& _a, U&& _b)
 {
 	std::move(_b.begin(), _b.end(), std::back_inserter(_a));
```

Comment:
Or in other words (if I'm right):
```suggestion
template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U&& _b)
{
	static_assert(!std::is_lvalue_reference_v<U>, "You should not be able to make this fail, because the U& version takes precedence!");
	std::move(_b.begin(), _b.end(), std::back_inserter(_a));
```

Reply:
I looks like that this is not working. It looks like that `U&&` version takes precedence.

Reply:
Ah no! You're right!